src/client/Client.cc

```diff
@@ -1152,6 +1155,13 @@ Inode * Client::add_update_inode(InodeStat *st, utime_t from,
     in->snap_caps |= st->cap.caps;
   }
 
+  if (need_snapdir_attr_refresh && in->is_dir() && in->snapid == CEPH_NOSNAP) {
```

Comment:
Should we also update this in https://github.com/ceph/ceph/pull/48086/commits/85fd356e6148561e453cef627878f113e3901446#diff-7a3052fe46aebfed0382c9d0bb9880ea1328add824e0b10c5d551ddfee282cd1R1067-R1070 ?

Reply:
in dump_status()?

Reply:
Sorry, I mean when the `mtime/ctime/atime` changed:

```
  if (new_version || (new_issued & CEPH_CAP_ANY_RD)) {
    update_inode_file_time(in, issued, st->time_warp_seq,
			   st->ctime, st->mtime, st->atime);
  }
```
Should we also set `need_snapdir_attr_refresh` to `true` ?

Reply:
iirc, that was already care in L#1041

Reply:
hmmm.. it doesn't hurt setting `need_snapdir_attr_refresh` here too TBH.